quantus/metrics/base.py

```diff
@@ -1,105 +1,331 @@
+            disable_warnings: bool = False,
+            display_progressbar: bool = False,
+            **kwargs):
         """
         Initialise the Metric base class.
 
         Each of the defined metrics in Quantus, inherits from Metric base class.
 
-        To add a new metric classes to the library, the minimum set of attributes that should be included are:
-
-            args: a arguments (optional)
-            kwargs: a dictionary of key, value pairs (optional)
-            abs: a bool stating if absolute operation should be taken on the attributions
-            normalise: a bool stating if the attributions should be normalised
-            normalise_func: Callable that make a normalising transformation of the attributions
-            default_plot_func: Callable that plots the metrics result
-            last_results: a list containing the resulting scores of the last metric instance call
-            all_results: a list containing the resulting scores of all the calls made on the metric instance
+        Parameters
+        ----------
+        abs (boolean): Indicates whether absolute operation is applied on the attribution, default=False.
+        normalise (boolean): Indicates whether normalise operation is applied on the attribution, default=True.
+        normalise_func (callable): Attribution normalisation function applied in case normalise=True.
+            If normalise_func=None, the default value is used, default=normalise_by_negative.
+        normalise_func_kwargs (dict): Keyword arguments to be passed to normalise_func on call, default={}.
+        softmax (boolean): Indicates wheter to use softmax probabilities or logits in model prediction, default=False.
+        default_plot_func (callable): Callable that plots the metrics result.
+        disable_warnings (boolean): Indicates whether the warnings are printed, default=False.
+        display_progressbar (boolean): Indicates whether a tqdm-progress-bar is printed, default=False.
 
         """
-        self.args = args
-        self.kwargs = kwargs
-        self.abs = self.kwargs.get("abs", False)
-        self.normalise = self.kwargs.get("normalise", False)
-        self.normalise_func = self.kwargs.get("normalise_func", normalise_by_negative)
-        self.default_plot_func = Callable
-        self.disable_warnings = self.kwargs.get("disable_warnings", False)
-        self.display_progressbar = self.kwargs.get("display_progressbar", False)
+        self.abs = abs
+        self.normalise = normalise
+        self.softmax = softmax
+
+        if normalise_func is None:
+            self.normalise_func = normalise_by_negative
+        self.normalise_func = normalise_func
+        if normalise_func_kwargs is None:
+            normalise_func_kwargs = {}
+        # TODO: deprecate this kind of unspecific kwargs passing
+        # this code prioritizes normalise_func_kwargs items over kwargs items
+        normalise_func_kwargs = {**kwargs, **normalise_func_kwargs}
```

Comment:
Can't we add:

```python
if kwargs:
    warn_funcs.kwargs_warning()
```
where 

```python

def kwargs_warning:
    warnings.warn(f"Passing **kwargs in the metric initalisation or call will be deprecated in future releases. Make sure to specify your parameters in the respective parameter space such as 'normalise_func_kwargs', 'explain_func_kwargs', 'model_predict_kwargs', 'perturb_func_kwargs' etc ..." " DeprecationWarning)

```

Reply:
Yes, this is a good idea to factor this out. I think we should also raise a warning instead of using print in `merge_kwargs`.